segmentstore/storage/impl/src/test/java/io/pravega/segmentstore/storage/impl/extendeds3/ExtendedS3StorageTest.java

```diff
@@ -73,6 +77,31 @@ public void tearDown() throws Exception {
         s3Proxy.stop();
     }
 
+    //region If-none-match test
+    /**
+     * Tests the create() method with if-none-match set.
+     */
+    @Test
+    public void testCreateIfNoneMatch() {
+        adapterConfig = ExtendedS3StorageConfig.builder()
+                                               .with(ExtendedS3StorageConfig.BUCKET, adapterConfig.getBucket())
+                                               .with(ExtendedS3StorageConfig.ACCESS_KEY_ID, "x")
+                                               .with(ExtendedS3StorageConfig.SECRET_KEY, "x")
+                                               .with(ExtendedS3StorageConfig.ROOT, "test")
+                                               .with(ExtendedS3StorageConfig.URI, endpoint)
+                                               .with(ExtendedS3StorageConfig.IFNONEMATCHWORKS, true)
+                                               .build();
+        String segmentName = "foo_open";
+        try (Storage s = createStorage()) {
+            s.initialize(DEFAULT_EPOCH);
+            s.create(segmentName, null).join();
+            assertThrows("create() did not throw for existing StreamSegment.",
```

Comment:
This code should throw independent of the value of `IFNONEMATCHWORKS`, right? I think the test case is useful in the sense that it checks that a second create should throw, but it doesn't look like it is testing the new option or even that the new flow doesn't break anything. Given that the unit tests run on the local file system, I wonder if the new option is making any difference in the processing of the put object request